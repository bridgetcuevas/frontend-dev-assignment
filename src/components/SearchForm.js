import React from "react";

const SearchForm = () => {
  return (
    <form action="" className="search-form" data-at="SearchForm">
      <input
        type="search"
        name="query"
        className="search-input"
        placeholder="Zoeken"
      />

      <button className="search-button" type="submit" />
    </form>
  );
};

export default SearchForm;
