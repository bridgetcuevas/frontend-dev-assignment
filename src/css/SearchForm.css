.search-form {
  margin: 50px;
  margin-right: auto;
  margin-left: auto;
  width: 270px;
}

.form-element {
  display: flex;
  align-items: flex-center;
  padding: 0.6rem;
  border: 1px solid #d8d8d8;
  -webkit-transition: border 0.3s cubic-bezier(0.23, 1, 0.32, 1);
  transition: border 0.3s cubic-bezier(0.23, 1, 0.32, 1);
}

.form-element:focus-within {
  border-color: #000;
}

.form-element,
.search-input {
  width: 100%;
  height: 100%;
  font-size: 13px;
  letter-spacing: 0.03em;
}

.search-input,
.search-button,
.clear-button {
  outline: none;
  border: none;
  overflow: hidden;
  background-color: #fff;
}

.clear-button {
  padding-right: 1em !important;
}

.search-button,
.clear-button {
  display: flex;
  align-items: center;
  padding: 0;
  margin: 0;
}

::placeholder {
  color: #777;
}

input[type="search"] {
  -webkit-appearance: textfield;
}

input[type="search"]::-webkit-search-cancel-button {
  -webkit-appearance: none;
}
