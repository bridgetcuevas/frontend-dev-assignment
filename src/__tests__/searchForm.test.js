import React from "react";
import SearchForm from "../components/searchForm";
import toJSON from "enzyme-to-json";
import { render } from "enzyme";

describe("Search input component", () => {
  it("should match snapshot", () => {
    const wrapper = render(<SearchForm />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
