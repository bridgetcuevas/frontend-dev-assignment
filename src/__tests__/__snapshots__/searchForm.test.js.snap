// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search input component should match snapshot 1`] = `
<form
  action=""
  class="search-form"
>
  <div
    class="form-element"
  >
    <input
      autocapitalize="off"
      autocomplete="off"
      autocorrect="off"
      class="search-input"
      name="query"
      placeholder="Zoeken"
      spellcheck="off"
      type="search"
    >
      
    </input>
    <button
      class="search-button"
      type="submit"
    >
      <svg
        class="sc-bdVaJa fUuvxv"
        fill="#777"
        height="17px"
        rotate="0"
        viewBox="0 0 1024 1024"
        width="17px"
      >
        <path
          d="M689 596c30-47.2 47.6-103.2 47.6-163.4 0-168.2-136.2-304.6-304.2-304.6-168.2 0-304.4 136.4-304.4 304.6s136.2 304.6 304.2 304.6c61 0 117.8-18 165.4-48.8l13.8-9.6 217.2 217.2 67.4-68.6-217-217.2 10-14.2zM602.8 262.4c45.4 45.4 70.4 105.8 70.4 170s-25 124.6-70.4 170c-45.4 45.4-105.8 70.4-170 70.4s-124.6-25-170-70.4c-45.4-45.4-70.4-105.8-70.4-170s25-124.6 70.4-170c45.4-45.4 105.8-70.4 170-70.4s124.6 25 170 70.4z"
        >
          
        </path>
      </svg>
    </button>
  </div>
</form>
`;
